Source/items.cpp

```diff
@@ -3889,14 +3889,13 @@ void UseItem(int p, item_misc_id Mid, spell_id spl)
 	switch (Mid) {
 	case IMISC_HEAL:
 	case IMISC_FOOD:
-		j = player._pMaxHP >> 8;
-		l = ((j >> 1) + GenerateRnd(j)) << 6;
+		l = GenerateRnd(player._pMaxHP / 4) + player._pMaxHP / 8;

```

Comment:
Is this intended to be included in this PR? This makes heals more granular, previously the low bits 5 bits would always be 0. It makes better use of the fixed point math anyway.

Reply:
https://github.com/diasurgical/devilutionX/pull/2055#discussion_r639110686
Good point about the 5 bits, the compiler should be fully capable of converting it to shifts though.

Reply:
Yeah, not worried about the divisions here. I *think* player._pMaxHP is always a value less than `2^15  - 1`, so using the current RNG with a higher value isn't going to alter the behaviour much.